rust/core-lib/src/tabs.rs

```diff
@@ -577,6 +578,14 @@ impl CoreState {
     // Detects whitespace settings from the file and merges them with the config
     fn detect_whitespace(&mut self, id: ViewId, config: &Table) -> Option<Table> {
         let buffer_id = self.views.get(&id).map(|v| v.borrow().get_buffer_id())?;
+        if self.editors.get(&buffer_id)
+            .map(|e| {
+                e.borrow().get_buffer().len() == 0
+            })
+            .unwrap_or_default() {
+            return None;
+        }
+
         let editor = self.editors.get(&buffer_id).expect("existing buffer_id must have corresponding editor");
```

Comment:
we already get the editor here; if we move the check down below this line, it's just:

```rust
if editor.get_buffer().is_empty() {
    return None;
}
```

Reply:
maybe you were thinking that this handles the case when there is no editor better?

Reply:
@YangKeao thoughts on this?

Reply:
@YangKeao 😁